actionpack/lib/action_controller/railtie.rb

```diff
@@ -86,5 +86,20 @@ class Railtie < Rails::Railtie #:nodoc:
         ActionController::Metal.descendants.each(&:action_methods) if config.eager_load
       end
     end
+
+    initializer "action_controller.query_log_tags" do |app|
+      ActiveSupport.on_load(:action_controller_base) do
+        mattr_accessor :query_log_tags_action_filter_enabled, instance_accessor: false, default: true
```

Comment:
I wonder if this might not be an issue because of how the module is structured?

```
irb(main):001:0> ActiveRecord::Base.connection.class::QueryLogTagsContext.ancestors
=> [ActiveRecord::ConnectionAdapters::QueryLogTags::QueryLogTagsContext]
```

Would love to know more about this problem in general though!

Reply:
Oh I just realised that's not the class in question.. `ActionController::Base` has a lot more. 

Sorry for noise, will look into alternatives / take any suggestions! :)

Reply:
Studying [this PR](https://github.com/rails/rails/pull/42445) for inspiration :)

Reply:
I _think_ I've resolved this in https://github.com/rails/rails/pull/42240/commits/d2ad44f1d1f6ade0037b66d7633f21eb48c71967 .

Reply:
Resolving as these have now all been updated to singleton attr_accessors.